tests/dragonfly/replication_test.py

```diff
@@ -1541,3 +1541,27 @@ async def test_df_crash_on_memcached_error(df_local_factory):
         memcached_client.set(b"key", b"data", noreply=False)
 
     await c_master.close()
+
+
+@pytest.mark.asyncio
+async def test_df_crash_on_replicaof_flag(df_local_factory):
+    master = df_local_factory.create(
+        port=BASE_PORT,
+        proactor_threads=2,
+    )
+
+    replica = df_local_factory.create(
+        port=master.port + 1, proactor_threads=2, replicaof=f"127.0.0.1:{BASE_PORT}"
+    )
+
+    master.start()
+    replica.start()
+
+    c_master = aioredis.Redis(port=master.port)
+    c_replica = aioredis.Redis(port=replica.port)
+
+    await wait_available_async(c_master)
+    await wait_available_async(c_replica)
+
+    res = await c_replica.execute_command(f"BGSAVE")
+    assert True == res
```

Comment:
I don't understand why we don't `OK` back here? @dranikpg any ideas?

Reply:
1. You don't assert in the test they're connected successfully, `wait_available_async` just waits till the loading phase is over (it can be absent at all). Please see in replication test, you can check replica status & set a key on master and check replica got it after wait available

2. 🤷🏻 Please see what it returns instead. It should indeed return OK if it saved successfully, I can't just tell by looking at the line of code 😆 But I can investigate if needed

Reply:
for (2), straight from the redis pylib:

```
**string_keys_to_dict("BGREWRITEAOF BGSAVE", lambda r: True),
```

This is the callback used to parse the result, so for some reason it transforms it to a `boolean` :rofl: 

1. I double checked by running the test against redis
2. I triple checked by putting a debugger on dragonfly and see that we call SendOk() + used the redis-cli to confirm this

> Please see in replication test, you can check replica status & set a key on master and check replica got it after wait available

It dosn't matter, if we got a result back it doesn't crash. Moreover, `BGSAVE` can't be called during `loading`, that's why I put it there....

Reply:
> It dosn't matter, if we got a result back it doesn't crash

Yes, I meant that the conditions for the bug were being a replica. I don't know it replicaof fails silently or not, either way I don't think it's almost impossible